R/CreateCohorts.R

```diff
@@ -6,133 +6,133 @@
 # limitations under the License.
 
 #' Builds SQL to create cohort table
-#' 
+#'
 #' @param sqlPath Path to SQL template
 #' @param resultsSchema The name of schema where cohort table will be placed
 #' @param targetTable The name of cohort table
-#' 
-#' 
+#'
+#'
 getCreateCohortTableSql <- function(sqlPath, resultSchema, targetTable) {
   createCohortTableSql <- SqlRender::readSql(sqlPath)
   createCohortTableSql <- SqlRender::render(createCohortTableSql, cohort_database_schema = resultSchema, cohort_table = targetTable)
   return(createCohortTableSql)
 }
 
 #' Builds SQLs to construct cohorts for analysis
-#' 
+#'
 #' @param studySpec Path to CC specification json
 #' @param cdmSchema The name of schema containing data in CDM format
 #' @param vocabularySchema The name of schema with vocabularies
 #' @param resultsSchema The name of schema where cohorts will be placed
 #' @param cohortTable The name of table where cohorts will be placed
-#' 
+#'
 getCohortSqls <- function(studySpec, cdmSchema, vocabularySchema, resultSchema, targetTable) {
   sqls <- c()
-  
+
   cohortExpression <- new(J("org.ohdsi.circe.cohortdefinition.CohortExpression"))
   queryBuilder <- new(J("org.ohdsi.circe.cohortdefinition.CohortExpressionQueryBuilder"))
-  
+
   cohorts <- fromJSON(studySpec)$cohorts
-  
+
   dbOptions <- list(
-    cdmSchema = cdmSchema,
-    targetTable = paste(resultSchema, ".", targetTable, sep = ""),
-    resultSchema = resultSchema,
-    vocabularySchema = vocabularySchema,
-    generateStats = FALSE
+  cdmSchema = cdmSchema,
+  targetTable = paste(resultSchema, ".", targetTable, sep = ""),
+  resultSchema = resultSchema,
+  vocabularySchema = vocabularySchema,
+  generateStats = FALSE
   )
-  
+
   for (c in cohorts) {
     options <- list(dbOptions)
-    
+
     options[[1]]$cohortId <- c$id
-    
+
     optionsJSON <- toJSON(options, container = FALSE)[[1]]
     queryExpressionOptions <- queryBuilder$BuildExpressionQueryOptions$fromJson(optionsJSON)
-    
+
     expressionJSON <- toJSON(c$expression)[[1]]
     expression <- cohortExpression$fromJson(expressionJSON)
-    
+
     sqls <- c(sqls, queryBuilder$buildExpressionQuery(expression, queryExpressionOptions))
   }
-  
+
   return(sqls)
 }
 
 #' Executes given SQLs to construct cohorts
-#' 
+#'
 #' @param cohortTableSql SQL to create cohort table
 #' @param cohortSqls Cohort SQLs
 #' @param connectionDetails An object of type \code{connectionDetails} as created using the
 #'                             \code{\link[DatabaseConnector]{createConnectionDetails}} function in the
 #'                             DatabaseConnector package.
-#' 
-constructCohorts <- function(cohortTableSql, cohortSqls, connectionDetails) {
-  
+#'
+constructCohorts <- function(cohortTableSql, cohortSqls, connectionDetails, tempSchema) {
```

Comment:
Could you add some default value for `tempSchema` like empty string or even `NULL` if possible?

Reply:
only two methods are "public"  and are called outside the file, so default value is not necessary here. 
`createCohorts <- function(pckg, connectionDetails, cdmSchema, vocabularySchema, resultSchema, tempSchema = resultSchema)`
`cleanupCohortTable <- function(connectionDetails, resultSchema, cohortTable, tempSchema = resultSchema`

and was tested locally :)